### PR TITLE
Documentation update

### LINKED ISSUE #50: Improve documentation
As discussed the documentation should include:
- more details about the interface
- a short example/demo without qibolab

### PR DESCRIPTION
**Pull request by rodolfocarobene**
Closes #50.
Still work in progress :-)

Checklist before merge:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Documentation is updated.

### CODE REVIEW

**Comment by rodolfocarobene** [comment]
This should now be ready, now the documentation looks as follow:
Introduction
├── Getting started
│   ├── Installation
│   ├── Usage
│   ├── Communication Protocol
│   ├── Pulses
│   ├── Sweepers
│   ├── Examples
│    │   ├── How to execute a pulse sequence
│    │   ├── How to execute a sweep
│    │   └── Example of a qubit spectroscopy
└── Api reference (generated with sphinx-apidoc)

A preview can be compiled locally with `make -C doc html`.

The last interesting thing that maybe we could add is the automatic sphinx-apidoc call inside the conf.py as proposed by @AleCandido in https://github.com/qiboteam/qibosoq/pull/43#discussion_r1236511785.
But I wasn't able to do it (the comment linked is outdated) and maybe it's not needed.

File: doc/source/getting-started/sweepers.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    server_commands = {
+        ...
+        sweepers: [asdict(sweeper), asdict(sweeper2)]
+    }
+
+This will execute the sequence considering the matrix product of the swept parameters.
+
+
+The final results (i and q) will have shape:
+* if averaged: (number_of_adc_chs, number_of_readouts, expts_sweeper1, expts_sweeper2...)
```
Comment (JavierSerranoGarcia) [inline doc/source/getting-started/sweepers.rst:70]: Write a proper list

File: doc/source/getting-started/communication.rst [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+* for operation_code ``EXECUTE_PULSE_SEQUENCE`` and ``EXECUTE_PULSE_SEQUENCE_RAW``
+    * if ``average`` is false: (adc_channels, number_of_readouts, number_of_shots)
+    * if ``average`` is true: (adc_channels, number_of_readouts)
+* for operation_code ``EXECUTE_SWEEPS``
+    * if ``average`` is false: (adc_channels, number_of_readouts, number_of_points, number_of_shots)
+    * if ``average`` is true: (adc_channels, number_of_readouts, number_of_points)
+
+Note that the server can also send a different thing: errors.
+Ehen the server encounters an error, in the communication protocol, in the json de-serialization or during the execution, it does not crash but raises an error that get's logged in the server and sent through the open socket so that also the client can see it.
```
Comment (JavierSerranoGarcia) [inline doc/source/getting-started/communication.rst:194]: Typo: Ehen -> When

**Review by JavierSerranoGarcia — APPROVED** [review]
Good

**Pull request merged**